adam/visualization/make_scenes.py

```diff
@@ -0,0 +1,156 @@
+from adam.perception.developmental_primitive_perception import (
+    DevelopmentalPrimitivePerceptionFrame,
+    RgbColorPerception,
+)
+
+from adam.visualization.panda3d_interface import SituationVisualizer
+from adam.visualization.utils import Shape
+
+USAGE_MESSAGE = """ """
+
+
+def main() -> None:
+    random.seed(2015)
+
+    # go through curriculum scenes (fed in from where?) and output geometry types
+    print("scene generation test")
+    sc = SceneCreator()
+    viz = SituationVisualizer()
+    for i, scene in enumerate(sc.create_scenes(GAILA_PHASE_1_CURRICULUM)):
+        print(f"SCENE {i}")
+        for obj, properties in scene.items():
+            if obj.geon is None:  # only interested in rendering geons
+                continue
+            # print(obj.geon.cross_section_size)
+            # print(obj.axes)
+            obj_added = False  # TODO: clean up this bit of control-flow mess
+            for prop in properties:
+                if isinstance(prop, RgbColorPerception):
+                    viz.add_model(
+                        sc.cross_section_to_geo(obj.geon.cross_section),
+                        SceneCreator.random_position(),
+                        prop,
+                    )
+                    obj_added = True
+            if not obj_added:
+                viz.add_model(
+                    sc.cross_section_to_geo(obj.geon.cross_section),
+                    SceneCreator.random_position(),
+                )
+        viz.run_for_seconds(2.25)
+        input("Press ENTER to continue")
+        viz.clear_scene()
+        viz.run_for_seconds(0.25)
+
+
+@attrs(frozen=True, slots=True)
+class SceneCreator:
+    def create_scenes(
+        self,
+        instance_groups: Iterable[
+            InstanceGroup[
+                HighLevelSemanticsSituation,
+                LinearizedDependencyTree,
+                DevelopmentalPrimitivePerceptionFrame,
+            ]
+        ],
+    ):
+        for i, instance_group in enumerate(
+            instance_groups
+        ):  # each InstanceGroup a page related to a curriculum topic
+            for (
+                situation,
+                dependency_tree,
+                perception,
+            ) in instance_group.instances():  # each instance a scene
+                # scene_objects = []
+                property_map = defaultdict(list)
+                # we only care about the perception at the moment
+
+                for frame in perception.frames:  # DevelopmentalPrimitivePerceptionFrame
+                    # print(frame.perceived_objects)
+                    # for obj_percept in frame.perceived_objects:
+                    #     if obj_percept.geon is None:
+                    #         continue
+                    #     scene_objects.append(
+                    #         self.cross_section_to_geo(obj_percept.geon.cross_section)
+                    #     )
+                    for property in frame.property_assertions:
```

Comment:
`property` is a built-in; [by convention](https://www.python.org/dev/peps/pep-0008/#function-and-method-arguments), use a name like` property_` instead.